### ISSUE TITLE
False Positive in UP007

### ISSUE THREAD

**Issue by Matt-Ord** [astral-sh/ruff#7452]
False positive when using Union of unpacked TypeVarTuple. As far as I know there is no way to represent this without using Union[].

```python
class Collection(Protocol[*_B0]):
    def __iter__(self) -> Iterator[Union[*_B0]]:
        ...
```
ruff 0.0.290

**Comment by charliermarsh** [comment]
Thanks!
